pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -211,4 +211,9 @@ private void closeSegment(SegmentDataManager segmentDataManager) {
   public String getTableName() {
     return _tableNameWithType;
   }
+
+  @Override
+  public String getTableDataDir() {
```

Comment:
Recommend returning an File to avoid the edge case of trailing `\`

Reply:
done.